server/src/rate_limit/rate_limiter.rs

```diff
@@ -0,0 +1,131 @@
+use super::*;
+
+#[derive(Debug, Clone)]
+pub struct RateLimitBucket {
+  last_checked: SystemTime,
+  allowance: f64,
```

Comment:
What exactly does this allowance mean? Would be good if you could add some comments to explain this stuff.

Reply:
Here's the pseudo code that I made this from: https://stackoverflow.com/questions/667508/whats-a-good-rate-limiting-algorithm